src/ensembl/src/shared/components/feature-summary-strip/GeneSummaryStrip.tsx

```diff
@@ -1,33 +1,69 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import classNames from 'classnames';
 
-import { BreakpointWidth } from 'src/global/globalConfig';
+import useResizeObserver from 'src/shared/hooks/useResizeObserver';
+
 import { getDisplayStableId } from 'src/shared/state/ens-object/ensObjectHelpers';
 import { getFormattedLocation } from 'src/shared/helpers/regionFormatter';
 
 import styles from './FeatureSummaryStrip.scss';
 
 import { EnsObject } from 'src/shared/state/ens-object/ensObjectTypes';
 
+const MEDIUM_WIDTH = 800;
+const SMALL_WIDTH = 500;
+
+enum Display {
+  FULL = 'full',
+  COMPACT = 'compact',
+  MINIMAL = 'minimal'
+}
+
 type Props = {
   gene: EnsObject;
   isGhosted?: boolean;
-  viewportWidth: BreakpointWidth;
 };
 
-const GeneSummaryStrip = ({ gene, isGhosted, viewportWidth }: Props) => {
+type WidthAwareProps = Props & {
+  display: Display;
+};
+
+const GeneSummaryWrapper = (props: Props) => {
+  const [display, setDisplay] = useState(Display.MINIMAL);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const { width: containerWidth } = useResizeObserver({ ref: containerRef });
+
+  useEffect(() => {
+    if (containerWidth < SMALL_WIDTH) {
+      display !== Display.MINIMAL && setDisplay(Display.MINIMAL);
```

Comment:
Should this statement have a `return`?

Reply:
Nope. We are using setDisplay to update the state.

Reply:
This is a bit confusing at first for me to spot. I wonder if it'll be the same for someone else :-/ Is it necessary to have the extra check to see if display is of the same type before setting it?

Reply:
I think so. We don't want to reset the state to the same value (i suspect that will cause rerendering).
We can have a nested if, but yuck!